desktop/ui/src/main/java/org/datacleaner/windows/HadoopConnectionToNamenodeDialog.java

```diff
@@ -0,0 +1,282 @@
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.windows;
+
+import java.awt.BorderLayout;
+import java.awt.GridBagLayout;
+import java.awt.Image;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import javax.swing.JButton;
+import javax.swing.JComponent;
+import javax.swing.JLabel;
+import javax.swing.event.DocumentEvent;
+
+import org.datacleaner.bootstrap.WindowContext;
+import org.datacleaner.configuration.ServerInformation;
+import org.datacleaner.panels.DCBannerPanel;
+import org.datacleaner.panels.DCPanel;
+import org.datacleaner.server.DirectConnectionHadoopClusterInformation;
+import org.datacleaner.user.MutableServerInformationCatalog;
+import org.datacleaner.util.DCDocumentListener;
+import org.datacleaner.util.ErrorUtils;
+import org.datacleaner.util.IconUtils;
+import org.datacleaner.util.ImageManager;
+import org.datacleaner.util.NumberDocument;
+import org.datacleaner.util.StringUtils;
+import org.datacleaner.util.WidgetFactory;
+import org.datacleaner.util.WidgetUtils;
+import org.datacleaner.widgets.Alignment;
+import org.datacleaner.widgets.DCLabel;
+import org.jdesktop.swingx.JXStatusBar;
+import org.jdesktop.swingx.JXTextField;
+
+/**
+ * Dialog for creating and editing a direct connection to Hadoop namenode
+ */
+public class HadoopConnectionToNamenodeDialog extends AbstractDialog {
+
+    private final DirectConnectionHadoopClusterInformation _directConnection;
+    private final JLabel _statusLabel;
+    private final JXTextField _nameTextField;
+    private final JXTextField _hostTextField;
+    private final JXTextField _portTextField;
+    private final JXTextField _descriptionTextField;
+    private final JButton _saveButton;
+    private final JButton _cancelButton;
+    private final MutableServerInformationCatalog _mutableServerInformationCatalog;
+    private ServerInformation _savedServer = null;
+    private static final ImageManager imageManager = ImageManager.get();
+
+    public HadoopConnectionToNamenodeDialog(WindowContext windowContext,
+            DirectConnectionHadoopClusterInformation directConnection,
+            MutableServerInformationCatalog serverinformationCatalog) {
+        super(windowContext);
+
+        _statusLabel = DCLabel.bright("Please specify connection name");
+        _directConnection = directConnection;
+        _nameTextField = WidgetFactory.createTextField("MyConnection");
+        _hostTextField = WidgetFactory.createTextField("localhost");
+        _portTextField = WidgetFactory.createTextField("9000");
```

Comment:
I know that we use 9000 on our test server, but the correct default namenode port is 8020.